wagtail/core/models.py

```diff
@@ -1379,6 +1476,22 @@ def copy(self, recursive=False, to=None, update_attrs=None, copy_revisions=True,
             page_copy.save()
 
         # Log
+        if log_action:
+            parent = specific_self.get_parent()
+            PageLogEntry.objects.log_action(
+                instance=page_copy,
+                action=log_action,
+                user=user,
+                data={
+                    'page': {
+                        'id': page_copy.id,
+                        'title': page_copy.get_admin_display_title()
+                    },
+                    'source': {'id': parent.id, 'title': parent.get_admin_display_title()} if parent else None,
+                    'destination': {'id': to.id, 'title': to.get_admin_display_title()} if to else None,
+                    'keep_live': page_copy.live and keep_live
```

Comment:
We discussed this previously and decided against it. The action is not copy & publish, but copy (and maintain published status if set). We could vary the displayed message since we have the metadata.

I'm in two minds about it. @gasman break the tie?

Reply:
I think _not_ having a separate log entry makes sense, since this is a log of 'actions performed', not 'state changes that happened as a result of actions'. If we were to start logging implied sub-actions in their own right, then logically you could make a similar case that copying a page also includes a 'create page' action, or that deleting a published page includes an 'unpublish' action, and it seems to me that going down that route would harm the usefulness of the logs.

Reply:
> I think not having a separate log entry makes sense, since this is a log of 'actions performed', not 'state changes that happened as a result of actions'.

When Dan and I discussed this, we came up with "micro-logging" where we make multiple logs for each action instead of trying to encode every possible combination of actions in Wagtail. This included making some actions more specific versions of others:

The Hierachy I'm thinking of is as follows:

```
wagtail.create
    wagtail.copy
        wagtail.copy_for_translation
    wagtail_content_import.import
    wagtail_transfer.import
wagtail.edit
    wagtail.revert
    wagtail_transfer.update
wagtail.publish
    wagtail.schedule.publish
wagtail.unpublish
    wagtail.schedule.expire
wagtail.delete
```

This heirachy would be encoded in the registry so that queries can be easily made to find all creations no matter what action was actually performed to create them.

If you copy a page and keep it live, I think it should emit both a `wagtail.copy` and a `wagtail.publish` log. This is to match how creating + publish is logged which emits a `wagtail.create` and `wagtail.publish` log

Reply:
As discussed this morning, added the publish log on copy. I've also updated the documentation to list the current actions and when they are logged + the rule of thumb for logging